### DISCUSSION TITLE
Add support for `useApi` in vue-router router guards

### DISCUSSION THREAD

**Discussion by TranQuangMinh** [directus/directus#14529] [Feature Requests]
I cant useApi in router of extension modules 

### To Reproduce

```js
export default defineModule({
	id: 'xxx',
	name: 'xxx',
	icon: 'box',
	routes: [
		{
			name: 'general',
			path: '',
			component: General,
			beforeEnter() {
				const api = useApi();
				console.log(api)
				// Error : [useApi]: The api could not be found.
			}
		}
	],
});
```

### Errors Shown

Error : [useApi]: The api could not be found.

**Comment by rijkvanzanten** [comment]
That's correct. The API is provided by the Vue app, and the router guard runs outside of the context of the vue component tree. I'll update this to be the feature request to allow for that usage (though I'm not sure if that's something we can support)

**Reply by nickrum** [reply]
Maybe we could allow App extensions to be defined by a function that returns the configuration object and pass anything that is exposed to extension components by provide/inject to this function:

```js
export default defineModule(({ api }) => ({
	id: 'xxx',
	name: 'xxx',
	icon: 'box',
	routes: [
		{
			name: 'general',
			path: '',
			component: General,
			beforeEnter() {
				console.log(api)
			}
		}
	],
}));
```

Though, we would have to make sure API, Stores etc. are available when registering extensions.

**Reply by rijkvanzanten** [reply]
Yeah that makes sense to me

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
